### DISCUSSION TITLE
Using custom postgresql database for Symfony 6.2 project

### DISCUSSION THREAD

**Discussion by hg2581** [symfony/symfony#51886] [Q&A] [answered]
New to Symfony and planning to build a sophisticated webapp centered around a custom postgresql database with many, many tables, relationships, constraints etc and fully utilizing the capabilities of postgresql which are not available through Doctrine.

What is the recommended way of integrating my database with the Symfony app and not be limited by Doctrine?

**Comment by tacman** [comment, marked as answer]
Use doctrine DBAL rather than the Doctrine ORM, and if that doesn't work, use raw queries.  See the SymfonyCasts series on doctrine, especially this chapter: https://symfonycasts.com/screencast/doctrine-queries/raw-sql-queries

**Reply by hg2581** [reply]
Thank you. I am new to Symfony so I have only looked at Doctrine ORM but DBAL and raw SQL queries sound very promising!
